### DISCUSSION TITLE
OTA via WiFi File upload in AP-mode

### DISCUSSION THREAD

**Discussion by Erhie** [easytarget/esp32-cam-webserver#263] [Show and tell]
Hello,
 I enhanced this software by adding the possibility for OTA update via WiFi with a binary file. This in softAP-mode .
 This is important for me because my project is located in free landscape without Internet and the steering device is an android tablet.
 As example I used code snippets from Jeija/esp32-softap-ota on GitHub.
I inserted
static esp_err_t update_post_handler_pre(httpd_req_t *req) {}
in app_httpd.cpp .
In this routine I started with 
  streamKill = true;
  delay(50);
  esp_camera_deinit();    
  periph_module_disable(PERIPH_I2C0_MODULE); // try to shut I2C down properly
  periph_module_disable(PERIPH_I2C1_MODULE);
  periph_module_reset(PERIPH_I2C0_MODULE);
  periph_module_reset(PERIPH_I2C1_MODULE);
and than call update_post_handler(req) in
main.c of the project Jeija/esp32-softap-ota.

**Comment by easytarget** [comment]
Thankyou for the suggestion; quite a reasonable idea, sorry I did not get round to it.

I am not doing any further development on this project. it is old code and somewhat obsolete. I am currently working on retiring it; I will try to link to a more actively updated fork when I do.
